pkgs/applications/networking/charles/default.nix

```diff
@@ -56,12 +58,16 @@ let
 
 in {
   charles4 = (generic {
```

Comment:
Can you extract this file into 3 files similar to generic.nix, 3.nix and 4.nix with the respective contents.